infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -108,6 +117,28 @@
     @Getter
     private volatile CompletableFuture<Boolean> sequencerBootstrappedFuture;
 
+    /**
+     * Timeout before which the management agent attempts to bootstrap a NOT_READY sequencer.
+     */
+    private static final int bootstrapNotReadySequencerTimeout = 3;
+
+    /**
+     * This tuple is maintained to track the un-bootstrapped primary sequencer.
+     * The epoch at which the primary sequencer was found NOT_READY and the timeout value.
+     */
+    private Map.Entry<Long, Integer> seqNotReadyCountAtEpoch;
+
+    /**
+     * Cluster metrics collection.
+     */
+    //  Service to poll local node metrics.
+    private final ScheduledExecutorService localMetricsPollingService;
```

Comment:
You might want to think about separating each of these functions into their own classes. It might make the code more readable.